src/sentry/static/sentry/app/components/bases/pluginComponentBase.jsx

```diff
@@ -99,13 +99,16 @@ class PluginComponentBase extends React.Component {
   }
 
   onSaveSuccess(callback, ...args) {
+    callback = callbackWithArgs(callback, ...args);
     this.setState(
       {
         state: FormState.READY,
       },
-      callbackWithArgs(callback, ...args)
+      () => callback && callback()
     );
-    addSuccessMessage(t('Success!'));
+    setTimeout(() => {
```

Comment:
@NisanthanNanthakumar Why are we using a `setTimeout` here? I don't see that being used in other places where we use `addSuccessMessage`?

Reply:
@scefali usually we put `addSuccessMessage` in the Promise handlers. In this case, there seems to be a race condition that forces the toasts to re-render quickly, causing them to disappear. This puts their execution to the back of the queue and lets them render for their default durations

Reply:
@NisanthanNanthakumar Can't we just call `addSuccessMessage` in the set state callback?

Reply:
@scefali `addErrorMessage` was originally inside the callback and it didn't render properly

Reply:
@NisanthanNanthakumar I really don't think adding a setTimeout here is the way to go. It's likely going to be a brittle solution. What if you move `addErrorMessage` to after the callback is called?

Reply:
@scefali I tried every combination/order and logged every interaction in the flow and this is the only solution that worked

Reply:
@scefali why do you think it’s brittle?

Reply:
@NisanthanNanthakumar Because it's a hack. But I don't see a better way...